src/workerd/io/worker-interface.capnp

```diff
@@ -32,12 +32,51 @@ struct Trace @0x8e8d911203762d34 {
     message @2 :Text;
   }
 
+  # Streaming tail workers support an expanded version of Log that supports arbitrary
+  # v8 serialized data or a text message. We define this as a separate new
+  # struct in order to avoid any possible non-backwards compatible disruption to anything
+  # using the existing Log struct in the original trace worker impl. The two structs are
+  # virtually identical with the exception that the message field can be v8 serialized data.
+  struct LogV2 {
+    timestampNs @0 :Int64;
+    logLevel @1 :Log.Level;
+    message :union {
+      # When data is used, the LogV2 message is expected to be a v8 serialized value.
+      data @2 :Data;
+      text @3 :Text;
+    }
+    tags @4 :List(Tag);
+    # A Log entry might be truncated if it exceeds the maximum size limit configured
+    # for the process. Truncation should occur before the data is serialized so it
+    # should always be possible to deserialize the data field successfully, regardless
+    # of the specific format of the data.
+    truncated @5 :Bool;
+  }
+
   exceptions @1 :List(Exception);
   struct Exception {
     timestampNs @0 :Int64;
     name @1 :Text;
     message @2 :Text;
     stack @3 :Text;
+
+    # Additional optional detail accompanying the exception event.
+    detail :group {
+      # If the exception has a cause property, it is serialized here.
+      cause @4 :Exception;
+
+      # If the exception represents an AggregateError or SupressedError, the
+      # errors are serialized here.
+      errors @5 :List(Exception);
+
+      # Additional metadata fields that are set on some errors originating
+      # from the runtime.
+      remote @6 :Bool;
+      retryable @7 :Bool;
+      overloaded @8 :Bool;
+      durableObjectReset @9 :Bool;
+      tags @10 :List(Tag);
```

Comment:
I'm considering replacing the `retryable`, `overloaded`, and `durableObjectReset` fields to be tags. However, the idea is to make it easier to add these additional pieces of metadata without schema changes later.

Reply:
> the idea is to make it easier to add these additional pieces of metadata without schema changes later.

We shouldn't be creating ways to avoid modifying the schema. People should just modify the schema as needed.

If we run into some case in the future where we see that modifying the schema is actually problematic, _then_ we can add some extensibility mechanism. But let's not design for problems that we don't yet have.